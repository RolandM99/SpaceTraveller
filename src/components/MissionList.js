import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import MissionsComponent from './MissionItem';
import { getMissionFromApi } from '../redux/reducers';

const MissionList = () => {
  const dispatch = useDispatch();
  const missions = useSelector((state) => state.reducers);

  useEffect(() => {
    if (!missions.missions[0]) {
      dispatch(getMissionFromApi());
    }
  }, []);
  return (
    <div className="missions">
      <table className="miss-table">
        <thead>
          <tr>
            <td className=" col-head">Mission</td>
            <td className=" col-head">Description</td>
            <td className=" col-head">Status</td>
            <td className=" col-head">Join</td>
          </tr>
        </thead>
        <tbody>
          {missions.missions.map((mission) => (
            <MissionsComponent key={mission.mission_id} mission={mission} />
          ))}
        </tbody>
      </table>
    </div>
  );
};

export default MissionList;
