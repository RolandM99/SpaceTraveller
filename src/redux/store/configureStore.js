import { createStore, combineReducers, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import rocketReducer from '../reducers';

const reducer = combineReducers({
  rocketReducer,
  // additional reducers could be added here
});

const store = createStore(
  reducer,
  applyMiddleware(logger, thunk),
);

export default store;
