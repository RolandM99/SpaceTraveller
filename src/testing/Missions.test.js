import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/store/configureStore';
import MissionList from '../components/MissionList';

describe('Mission Component test', () => {
  it('Renders Mission Row', () => {
    const app = renderer
      .create(
        <Provider store={store}>
          <MissionList />
        </Provider>,
      )
      .toJSON();
    expect(app).toMatchSnapshot();
  });
});
