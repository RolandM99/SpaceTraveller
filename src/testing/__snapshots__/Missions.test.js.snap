// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mission Component test Renders Mission Row 1`] = `
<div
  className="missions"
>
  <table
    className="miss-table"
  >
    <thead>
      <tr>
        <td
          className=" col-head"
        >
          Mission
        </td>
        <td
          className=" col-head"
        >
          Description
        </td>
        <td
          className=" col-head"
        >
          Status
        </td>
        <td
          className=" col-head"
        >
          Join
        </td>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
